### PR TITLE
Updated object to class name #57

### LINKED ISSUE #57: `obj = object.__new__(cls)`
#### Description of the problem

Currently any class object is created by,
```python
obj = object.__new__(cls)
```
For better readability, each such line should be replaced by the abstract superclass of that class.
Something like,
```python
obj = <name_of_abstract_super_class>.__new__(cls)
```
For example, in `arrays.py`,  `OneDimensionalArray` is created by, 
https://github.com/codezonediitj/pydatastructs/blob/9c32d4068ace96da9b953ea678b1c2fd852d141c/pydatastructs/linear_data_structures/arrays.py#L73
However, the above should be replaced by,
```python
obj = Array.__new__(cls)
```

### PR DESCRIPTION
**Pull request by TarunTomar122**
fixes #57

#### Brief description of what is fixed or changed
Replaced lines having object.__new__ to base class name __new__..

### CODE REVIEW

File: pydatastructs/miscellaneous_data_structures/stack.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def __new__(cls, maxsize=None, top=0, items=None, dtype=int):
         if items._size > maxsize:
             raise ValueError("Overflow, size of items %s is greater "
                             "than maxsize, %s"%(items._size, maxsize))
-        obj = object.__new__(cls)
+        obj = Stack.__new__(cls)
```
Comment (czgdp1807) [inline pydatastructs/miscellaneous_data_structures/stack.py]: We have to keep it as `object`. This is an exception to the generic fix. Rest is fine.

Comment (TarunTomar122) [reply pydatastructs/miscellaneous_data_structures/stack.py]: Done!!

**Comment by czgdp1807** [comment]
Make the changes at the following place as well,

https://github.com/codezonediitj/pydatastructs/blob/27d3ed84602f157e889ad7d3e25a10af10750700/pydatastructs/trees/heaps.py#L70

**Comment by TarunTomar122** [comment]
But Binary heap is not inheriting any other class than object class itself!

**Comment by czgdp1807** [comment]
Well, then we need to make it inherit, `Heap` and then make the change. In future, there may be a need to shift some common methods to the topmost class.

**Comment by czgdp1807** [comment]
Thanks. Merging.

**Pull request merged**